test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,347 @@
+        {
+          env: undefined,
+          client: undefined,
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stderr',
+          client: undefined,
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stdout',
+          client: undefined,
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: undefined,
+          client: 'stdout',
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: 'stderr',
+          client: 'stdout',
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: 'stdout',
+          client: 'stdout',
+          expectedLogDestination: process.stdout
+        },
+        {
+          env: undefined,
+          client: 'stderr',
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stderr',
+          client: 'stderr',
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: 'stdout',
+          client: 'stderr',
+          expectedLogDestination: process.stderr
+        },
+        {
+          env: undefined,
+          client: stream,
+          expectedLogDestination: stream
+        },
+        {
+          env: 'stderr',
+          client: stream,
+          expectedLogDestination: stream
+        },
+        {
+          env: 'stdout',
+          client: stream,
+          expectedLogDestination: stream
+        }
+      ];
+
+      for (const { env, client, expectedLogDestination } of tests) {
+        context(
+          `environment option=${env}, client option=${
+            client instanceof Writable ? 'a writable stream' : client
+          }`,
+          () => {
+            it(`sets the log destination to ${
+              expectedLogDestination instanceof Writable
+                ? 'the provided writable stream'
+                : expectedLogDestination
+            }`, () => {
+              const options = MongoLogger.resolveOptions(
+                { MONGODB_LOG_PATH: env },
+                { mongodbLogPath: client }
+              );
+
+              expect(options).to.have.property('logDestination', expectedLogDestination);
+            });
+          }
+        );
+      }
+    });
+  });
+
+  describe('severity helpers', function () {
+    const severities = Object.values(SeverityLevel).filter(severity => severity !== 'off');
+    for (const severityLevel of severities) {
+      describe(`${severityLevel}()`, function () {
+        it('does not log when logging for the component is disabled', () => {
+          const stream = new BufferingStream();
+          const logger = new MongoLogger({
+            componentSeverities: {
+              topology: 'off'
+            } as any,
+            logDestination: stream
+          } as any);
+
+          logger[severityLevel]('topology', 'message');
+          expect(stream.buffer).to.have.lengthOf(0);
```

Comment:
quick question, do we expect the actual call to write to happen synchronously here? (I see the meta-tests above to ensure that the buffer is filled synchronously, but it may well be possible that we have some async steps between the call to the logger and the call to the stream.write) if not, could be worth adding a timeout of some sort to give the event loop some time to process (basically, when we do implement the methods, I would expect the same logic to be able to be used in the positive cases)

Reply:
i thought about that, but decided against it because 

1. i foresee no timeouts / async work in the logging scope
2. if there is, who knows how long of a timeout is sufficient now?  we don't know the extent of the async work

but I'm happy to adjust this